### ISSUE TITLE
Layout and Appearance

### ISSUE THREAD

**Issue by wtheisen** [wtheisen/commandChan#9]
@sambattalio and @ginglis13 what do you guys think of the layout of the views. I’m starting to think that we could make a much better and more useful looking layout. Right now overlaying in a single pane looks nice but it doesn’t provide any extra utility. what are your thoughts on an appearance rewrite?

**Comment by wtheisen** [comment]
As an aside I'd probably write all the code myself I just want opinions on what makes sense

**Comment by sambattalio** [comment]
I think an appearance rewrite with tree view in mind would be definitely useful. I definitely agree on doing a rewrite.

**Comment by ginglis13** [comment]
agreed yea tree view would be nice. maybe whenever you go a level deeper, the whole screen clears to just show that level, with a tree view of the levels to the side? or maybe even post names to the side? idk. def think a layout rewrite could be cool

**Comment by wtheisen** [comment]
So I was thinking of trying to further emulate vim by having an idea of view that would display frames.
Where a view would be like a split in vim and a frame would be a proxy for a file. So we could have a history frame, a thread frame, board frame etc. etc. Per the reddit idea I agree it has to be a tree unless we decided to override the LineBox class to add indentation levels to it

**Comment by wtheisen** [comment]
I started working on the split code yesterday and I made a little progress but it's going to be tricky

**Comment by wtheisen** [comment]
Oh yeah i also think there's no reason to have it continue being an overlay object, it looks nice but it doesn't add anything other than that.

**Comment by sambattalio** [comment]
I like the vim emulation idea. If you need help with tree view for reddit I can definitely try to take the charge on it if you need (since I assume you might prefer working on 4chan over reddit)

**Comment by wtheisen** [comment]
http://urwid.org/examples/index.html

There is a browse.py example that implements a tree view with collapsing and expanding

**Comment by sambattalio** [comment]
aight i got u thanks

**Comment by noyoshi** [comment]
I think that doing something like highlighting important text like user names, subreddit / board names, etc, might be nice, that way you can easily see whats on the default frame when you load up a view.

**Comment by wtheisen** [comment]
I also think we need to add some sort of bookmarking ability. This could probably be done really easily using the already existing history list/Frame
